doc/centos-tutorial.txt

```diff
@@ -0,0 +1,141 @@
+
+B. LAUNCHING, CHECKPOINTING AND RESTARTING A MPI APPLICATION
+
+We'll first add the bin directory to PATH. Ensure that these commands are done
+from within the root directory of MANA.
+
+  $ echo 'export PATH=$PATH:{PATH TO MANA}/bin' >> ~/.bashrc
+  $ source ~/.bashrc
+
+By default, this should be /home/{YOUR USERNAME}/mana.
+
+1. Launching an MPI application
+
+The MANA directory comes with many test MPI applications that can be found in
+contrib/mpi-proxy-plugin/test. Depending on the application, we may require
+more than 1 copy of the process running - for example, ping_pong.mana.exe
+requires 2. To support this, change the argument after -np accordingly. For this
+tutorial, we'll use mpi_hello_world.mana.exe, which only requires 1 copy
+running.
+
+  $ mana_coordinator
+  $ mpirun -np 1 mana_launch \ 
+      contrib/mpi-proxy-split/test/mpi_hello_world.mana.exe
+
+If the application is launched properly, you should see the following printed:
+
+  Hello world from processor test, rank 0 out of 1 processors
+  Will now sleep for 500 seconds ...
+
+From here, we can either exit the program (through CTRL-C), or continue on to
+the next set of instructions to checkpoint and restart the application.
+
+2. Checkpointing and restarting a MPI application
+
+Currently, the fix for mana_restart has not been pushed to the feature/centos
+branch yet; it's going through some testing in the final phases, but we'll
+simply grab the pull request and patch it in.
+
+  $ wget https://github.com/mpickpt/mana/pull/29.patch
+  $ git apply 29.patch
+  $ make -j mana
+
+Here, we want to either open another terminal, or have the launched MPI
+application be open in the background. The following demonstrates the latter.
+
+If an application is already launched, we do CTRL-Z, followed by:
+
+  $ bg
+
+Otherwise, we'll launch an application the same way as above, but as a
+background process (note the &).
+
+  $ mana_coordinator
+  $ mpirun -np 1 mana_launch \ 
+      contrib/mpi-proxy-split/test/mpi_hello_world.mana.exe &
+
+Subsequently, we only need the following command to checkpoint the application.
+
+  $ mana_status --checkpoint
+
+To restart the application, we do the following:
+
+  $ mana_coordinator
+  $ mpirun -np 1 mana_restart
+
+Depending on whether there is another instance of mana_coordinator running, we
+may need to either close all other instances or launch a coordinator with a
+port different from the default 7779.
+
+  $ mana_coordinator -p 7780
+
+If the restart is successful, then you should see something similar to the
+following printed:
+
+  [17847] mtcp_restart.c:803 main:
+    [Rank: 0] Choosing ckpt image: ./ckpt_rank_0/ckpt_mpi_hello_world.mana.exe_
+      7c4af7b7e161f9fd-40000-1000f911e89fa5.dmtcp
+  Signal received; continuing sleep for 294 seconds.
+
+C. MISCELLANEOUS RESOURCES
+
+For more information on MANA, you may refer to the manpage in the directory.
+From the root MANA directory, do:
+
+  $ nroff -man manpages/mana.1
```

Comment:
And maybe add:

>D. CAVEATS
At the time of this writing, runtime overhead can be significant for MPI applications that make very frequent calls to MPI.  This is a known issue, and future versions will have lower runtime overhead.